pkg/provision/provision.go

```diff
@@ -295,10 +295,11 @@ func updateUnit(p provision.SSHCommander, name string, content string, dst strin
 	glog.Infof("Updating %s unit: %s ...", name, dst)
 
 	if _, err := p.SSHCommand(fmt.Sprintf("sudo mkdir -p %s && printf %%s \"%s\" | sudo tee %s.new", path.Dir(dst), content, dst)); err != nil {
-		return err
+		return errors.Wrapf(err, "generating %q system unit", name)
 	}
-	if _, err := p.SSHCommand(fmt.Sprintf("sudo diff -u %s %s.new || { sudo mv %s.new %s; sudo systemctl -f daemon-reload && sudo sudo systemctl -f restart %s; }", dst, dst, dst, dst, name)); err != nil {
-		return err
+	cmd := fmt.Sprintf("sudo diff -u %s %s.new || { sudo mv %s.new %s; sudo systemctl -f daemon-reload && sudo sudo systemctl -f restart %s; }", dst, dst, dst, dst, name)
+	if _, err := p.SSHCommand(cmd); err != nil {
+		return errors.Wrapf(err, "running %q", cmd)
```

Comment:
Doesn't the error already include cmd?

Reply:
No. I believe this command goes through a different route